db/00_init.sql

```diff
@@ -254,3 +254,60 @@ BEGIN
 END;
 $$ LANGUAGE plpgsql
 IMMUTABLE LEAKPROOF ROWS 100;
+
+-- Statistics per project and administrative boundary
+CREATE TABLE pdm_feature_counts_per_boundary(
+	project VARCHAR NOT NULL,
+	boundary BIGINT NOT NULL,
+	day DATE NOT NULL,
+	amount INT NOT NULL,
+
+	CONSTRAINT pdm_feature_counts_per_boundary_pk PRIMARY KEY(project, boundary, day)
+);
+
+CREATE INDEX pdm_feature_counts_per_boundary_project_idx ON pdm_feature_counts_per_boundary(project);
+
+-- Count edits per administrative boundary
+CREATE OR REPLACE FUNCTION pdm_add_changes_per_boundary(project_id VARCHAR, start_ts TIMESTAMP, end_ts TIMESTAMP) RETURNS VOID AS $$
+DECLARE
+	curr_day DATE;
+BEGIN
+	curr_day := start_ts::DATE;
+
+	-- Cleanup before counting
+	DELETE FROM pdm_feature_counts_per_boundary
+	WHERE project = project_id AND day BETWEEN start_ts::DATE AND end_ts;
+
+	-- List of edits in each boundary
+	EXECUTE format('
+		CREATE TABLE pdm_boundary_edits_tmp AS
+		SELECT b.osm_id, b.name, f.day AS edit_day
+		FROM (
+			SELECT DISTINCT ON (c.osmid) c.osmid, c.ts::date AS day, c.action, p.geom
+			FROM pdm_changes c
+			JOIN %I p ON c.osmid = p.osm_id
+			WHERE c.project = %L AND c.ts BETWEEN %L AND %L
```

Comment:
Je vois qu'on utilise la table des change mais cherche-t-on à compter les créations, les éditions ou les deux ?

Cela m'étonne qu'il n'y ait pas de filtre sur le champ `contrib`

Reply:
Au début je ne m'étais pas posé la question, et en cours de route je me suis aperçu du problème pour tout ce qui est suppression. Mais en y repensant, comme l'objet de cette carte est de localiser où sont les contributeurs (et donc où sont les zones où il y a eu peu de participation), c'est pas tant le nombre d'objets qui compte que l'activité. Du coup compter le nombre d'objets "édités" au sens large (créé, modifié, supprimé, mais chaque objet n'est compté qu'une fois même s'il est édité plusieurs fois) n'est pas si mal. Et l'avantage c'est que l'on peut retracer l'historique jour par jour pour chaque zone sans que ce soit plus compliqué en terme de traitements.

Reply:
> créé, modifié, supprimé, mais chaque objet n'est compté qu'une fois même s'il est édité plusieurs fois

merci pour les infos.
Tu as raison, le group by ligne 304 ne donne qu'un compte par objet.
Il serait certainement possible d'arriver simplement à prendre en compte les suppressions en déterminant le start et end via un switch case ayant respectivement pour valeur par défaut le début du projet et la date du jour.
Ensuite la ligne 303 se transformer en un `WHERE curr_day BETWEEN start AND end`

Reply:
Après si on a des projets où l'on cherche à supprimer des objets, pouvoir penser nombre contributions que nombre d'objets ce sera pas plus mal.